sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -324,6 +328,41 @@ class RapidsExecutorPlugin extends ExecutorPlugin with Logging {
     }
   }
 
+  // Wait for command spawned via Process
+  private def waitForProcess(cmd: Process, durationMs: Long): Option[Int] = {
+    val endTime = System.currentTimeMillis() + durationMs
+    do {
+      Thread.sleep(10)
```

Comment:
good old Java allows to waitFor with timeout https://docs.oracle.com/javase/8/docs/api/java/lang/Process.html#waitFor-long-java.util.concurrent.TimeUnit-